db/db.go

```diff
@@ -68,19 +64,58 @@ type DB struct {
 
 // NewDB creates a new DB, which will *own* ds and dispatcher for internal use.
 // Saying it differently, ds and dispatcher shouldn't be used externally.
-func NewDB(ts service.Service, opts ...Option) (*DB, error) {
+func NewDB(ctx context.Context, ts s.Service, id thread.ID, opts ...Option) (*DB, error) {
 	config := &Config{}
 	for _, opt := range opts {
 		if err := opt(config); err != nil {
 			return nil, err
 		}
 	}
-	return newDB(ts, config)
+
+	ti, err := ts.GetThread(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if len(ti.Logs) == 0 {
+		ti, err = ts.CreateThread(ctx, id, s.FollowKey(sym.New()), s.ReadKey(sym.New()))
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Well, `GetThread` does not error in the case where we have no info about a thread. That method pulls info from all the log books. We could certainly make it error if all the keys _and_ logs are empty.

Reply:
Yeah, my point was that the: _get X, don't have an error, then create X_ feels weird without knowing the implementation details. Mostly, from a client-side perspective. 
Feels like the `CreateThread` method call should be called `CreateThreadLog` or similar. 

Not sure we should do something about it now, but sharing this since I haven't used the APIs for some time and this doubt bubbled up.